### PR TITLE
Get the README in shape for initial release

### PR DESCRIPTION
**Pull request by tomhoule**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,44 +1,57 @@
...
+![CI Status](https://img.shields.io/travis/tomhoule/graphql-client.svg)
+![docs](https://docs.rs/mio/badge.svg)
+![crates.io](https://img.shields.io/crates/d/graphql-client.svg)
+![license](https://img.shields.io/github/license/mashape/apistatus.svg)
+![PRs Welcome](https://img.shields.io/badge/PRs-welcome-brightgreen.svg)
 
-This library does not provide any networking, caching or other client functionality, it is just meant to make it easy to interact with a GraphQL query and the corresponding response in a strongly typed way. Building a client can be as simple as this:
+Derive Rust code to safely and ergonomically manipulate GraphQL queries.
+
+This library does not provide any networking, caching or other client functionality yet, it is just meant to make it easy to interact with a GraphQL query and the corresponding response in a strongly typed way. Making a request can be as simple as this:
```
Comment (brainlessdeveloper) [inline README.md]: I'd remove the `yet`, it leaves the intention of the library and its roadmap unclear.

Comment (brainlessdeveloper) [reply README.md]: This paragraph has the purpose of saying what the library does/does not, but you also introduce the recipe with the last sentence. The last sentence should be in a new line and in bold (IMO).

Comment (brainlessdeveloper) [reply README.md]: I think the purpose of the library (i.e. what it _does_) should go closer to the tagline "Derive Rust code..." and the "what it _doesn't_ do" should go elsewhere, maybe after the features section, and it should be visually different (italics or something).

Comment (brainlessdeveloper) [reply README.md]: The fact that it doesn't provide networking, caching or other functionalities is not part of the "purpose" of the library (apparently) so I think it should be mentioned a lot less prominently.

File: README.md [hunk 2]
```diff
@@ -1,44 +1,57 @@
...
+    schema_path = "src/graphql/schema.json"
 )]
 struct MyQuery;
 
 fn perform_my_query(variables: &my_query::Variables) -> Result<(), failure::Error> {
     let body = MyQuery::expand(variables);
     let client = reqwest::Client::new();
-    let res: HttpResponse<graphql_client::Response<my_query::ResponseData>> = client.post("/graphql", body)?;
-    println!("{:#?}", res.body);
+    let mut res: HttpResponse<graphql_client::Response<my_query::ResponseData>> = client.post("/graphql").json(&body).send()?;
```
Comment (brainlessdeveloper) [inline README.md]: This line doesn't fit inside GitHub's README width, maybe you could split it?

Also, I'd add a newline after the `let mut res` line.

File: README.md [hunk 3]
```diff
@@ -1,44 +1,57 @@
...
+    println!("{:#?}", res.text());
+    Ok(())
 }
 ```
 
+The paths in the `gql` attribute are relative to the directory where your `Cargo.toml` is located.
```
Comment (peterfication) [inline README.md]: I would move this in the code example as comment

File: README.md [hunk 4]
```diff
@@ -1,44 +1,57 @@
...
+
+The GraphQL schema language and schema.json are both supported as sources for the schema.
+
+`serde_derive` needs to be visible in the context of the `GraphQLQuery` derive (add it as an `extern crate`).
```
Comment (brainlessdeveloper) [inline README.md]: These three lines are out of context and (I think) should be under a headline, something like "Set-up", where you say what a user of the library has to do to get it working. These three lines look like they belong in the details of such a section.

File: README.md [hunk 5]
```diff
@@ -1,44 +1,57 @@
...
-    println!("{:#?}", res.body);
+    let mut res: HttpResponse<graphql_client::Response<my_query::ResponseData>> = client.post("/graphql").json(&body).send()?;
+    println!("{:#?}", res.text());
+    Ok(())
 }
 ```
 
+The paths in the `gql` attribute are relative to the directory where your `Cargo.toml` is located.
+
+The GraphQL schema language and schema.json are both supported as sources for the schema.
```
Comment (peterfication) [inline README.md]: schema.json => `schema.json`

File: README.md [hunk 6]
```diff
@@ -1,44 +1,57 @@
...
+
+`serde_derive` needs to be visible in the context of the `GraphQLQuery` derive (add it as an `extern crate`).
+
 ## Features
```
Comment (brainlessdeveloper) [inline README.md]: I would put the whole features section a little bit more towards the start, maybe right after the recipe.

File: README.md [hunk 7]
```diff
@@ -1,44 +1,57 @@
...
-* An impl for the `GraphQLQuery` trait for the struct under derive
+- A module named after the struct under derive, which contains:
+  - A `ResponseData` struct implementing `serde::Deserialize`
+  - A `Variables` struct meant to contain the variables expected by the query
+- An impl for the `GraphQLQuery` trait for the struct under derive
 
-See the example generated module for a full example.
+See the [example generated module](https://www.tomhoule.com/docs/example_module/) for more details.
 
 ## Examples
```
Comment (peterfication) [inline README.md]: I would add a link to the folder in the text

File: README.md [hunk 8]
```diff
@@ -1,44 +1,57 @@
...
+
 ## Features
 
-* Strongly typed query variables
-* Strongly typed response
+- Strongly typed query variables
+- Strongly typed responses
+- Works in the browser (WebAssembly)
+
+Integration with different HTTP libraries is planned, although building one yourself is trivial (just send the constructed request payload as JSON with a POST request to a GraphQL endpoint, modulo authentication).
```
Comment (brainlessdeveloper) [inline README.md]: The line about planned features and the CLI should be in a smaller "Roadmap" headline under features (IMO).

File: graphql_client_cli/README.md [hunk 9]
```diff
@@ -1 +1,3 @@
-https://developer.deutschebahn.com/free1bahnql/graphql
+# GraphQL client CLI
+
+This is still a WIP, the main use for it now is to download the schema.json from a GraphQL endpoint, which you can also do with [apollo-codegen](https://github.com/apollographql/apollo-cli).
```
Comment (peterfication) [inline graphql_client_cli/README.md]: schema.json => `schema.json`

File: README.md (hunk 7 above)
Comment (brainlessdeveloper) [inline README.md]: The "examples" line after this could include a link.

File: README.md (hunk 3 above)
Comment (tomhoule) [reply README.md]: good idea!

File: README.md (hunk 7 above)
Comment (brainlessdeveloper) [reply README.md]: Maybe also some text directly to specific examples explaining something about them.

File: README.md [hunk 10]
```diff
@@ -1,45 +1,79 @@
...
+See the [example generated module](https://www.tomhoule.com/docs/example_module/) for more details.
 
 ## Examples
 
 See the examples directory in this project.
+
+## Code of conduct
+
+Anyone who interacts with this project in any space, including but not limited to
+this GitHub repository, must follow our [code of conduct](https://github.com/tomhoule/graphql-client/blob/master/CODE_OF CONDUCT.md).
```
Comment (indifferentalex) [inline README.md]: Link is broken because of the missing underscore

Comment (tomhoule) [reply README.md]: thanks!

**Comment by tomhoule** [comment]
I'll keep the comments on this PR in mind for the rest of the work on the README.

**Pull request merged**